src/main/java/vivatech/util/TierHelper.java

```diff
@@ -15,17 +15,10 @@ public static Identifier getTieredID(Identifier id, MachineTier tier) {
 		return new Identifier(id.getNamespace(), tier.getAffix() + "_" + id.getPath());
 	}
 	
-	public static AbstractTieredMachineBlock[] fillTieredBlockArray(Class<? extends AbstractTieredMachineBlock> clazz, AbstractTieredMachineBlock[] array) {
-		for(MachineTier tier : MachineTier.values()) {
-			try {
-				array[tier.ordinal()] = clazz.getConstructor(MachineTier.class).newInstance(tier);
-			} catch (InstantiationException | IllegalAccessException | IllegalArgumentException
-					| InvocationTargetException | NoSuchMethodException | SecurityException e) {
-				e.printStackTrace();
-				System.exit(1);
-			}
-		}
-		return array;
+	public static void fillTieredBlockArray(AbstractTieredMachineBlock[] blocks, Function<MachineTier, ? extends AbstractTieredMachineBlock> function) {
+	    for(int i = 0; i < MachineTier.values().length; i++) {
```

Comment:
When the array is declared up top, and passed in, it means there's a slight chance that a typo could send you into an `ArrayIndexOutOfBoundsException`, which is why I created the array here in the helper method. This is still entirely reasonable how you have it, I'm just sharing my rationale.

Reply:
I would prefer to create the array in the helper method but java crashes with `The blank final field ELECTRIC_FURNACE may not have been initialized` if I do it in the helper.

Reply:
fair enough, we don't have kotlin lateinit, so if you went with my plan you'd have to ditch final.

Reply:
Yeah, that would work but it also goes against the current system of initializing everything in `static {}` and I've had some trouble with casting arrays up. Not sure how much of an issue either of those would be.

Reply:
I don’t really understand why do you need to ditch `final`. There’s nothing that requires some code initialized after finals

Reply:
We only have to if we want to avoid initializing and filling the array separately. Currently because Java crashes if it's not sure final variables are set and I can't cast `AbstractTieredMachineBlock[]` to an array of Electric furnace blocks I have to initializing the array in the class and then fill it so Java doesn't crash.

Reply:
Then why don’t change the field to other type?

Reply:
Yeah, we could change tiered blocks to all be arrays of AbstractTieredMachineBlock. As long as doing so won't upset any of your plans I can make the change now.

Reply:
Do it. I can manage it later if I need to, so np

Reply:
Great, done.